### ISSUE TITLE
[BUG] user_followers gets an only 249 items

### ISSUE THREAD

**Issue by zhito** [adw0rd/instagrapi#1318]
Hello!
mb something with pagination
instagrapi==1.17.6
cl.user_followers(inst_pk)
cl.user_followers(inst_pk, False, amount=2000)

**Comment by adw0rd** [comment]
try `cl.user_followers_gql`

**Comment by zhito** [comment]
the same:
user_followers = self.cl.user_followers_gql(inst_pk)
logging.info(len(user_followers))
2023-05-05  [INFO ]  249

**Comment by adw0rd** [comment]
@zhito show inst_pk

**Comment by mdczaplicki** [comment]
Same here. These functions return only 288 followers, even though there ought to be over 40k.

User id is: `45600199849`.

**Comment by adw0rd** [comment]
@zhito @mdczaplicki 

Unfortunately, I don’t have time to look for reasons in instgrapi right now, but it also works on Lamadava on it and everything looks ok:


![image](https://github.com/adw0rd/instagrapi/assets/546889/7eca7b15-ffc5-49fd-8a4c-a515991f6854)


<img width="782" alt="image" src="https://github.com/adw0rd/instagrapi/assets/546889/67575fdc-4965-4130-bc92-39a466b57201">

**Comment by mdczaplicki** [comment]
Number of returned items varies. Now (after some time) it returns 300 records. Consecutive requests return empty list.
```
In [2]: followers = client.user_followers_gql(45600199849)

In [3]: len(followers)
Out[3]: 300

In [4]: 

In [4]: followers = client.user_followers_gql(45600199849, 500)

In [5]: len(followers)
Out[5]: 0

In [6]: followers = client.user_followers_gql(45600199849, amount=500)

In [7]: len(followers)
Out[7]: 0
```

**Comment by claell** [comment]
I think I have read about this before. There it was thought to be some limit from Instagram's side.

**Comment by LiveMato** [comment]
> Same here. These functions return only 288 followers, even though there ought to be over 40k.
> 
> User id is: `45600199849`.

Yes, I have same problem, but now i use cl.user_followers_v1(inst_pk, amount=2000) and I can get latest 2k followers, but with cl.user_followers_gql(inst_pk), i get just 300 followers :/

**Comment by Macerul** [comment]
> the same: user_followers = self.cl.user_followers_gql(inst_pk) logging.info(len(user_followers)) 2023-05-05 [INFO ] 249

I'm trying to do the same (looping into a follower list and saving public info like number of media, followers etc) but I have two questions:
-doest instagram blocks you after retrieving just 249 items? If it doesn't, will you be able to retrieve the others running the code in another moment?

**Comment by diezo** [comment]
I managed to fetch over 10k followers (then I stopped the script) using [Ensta](https://github.com/diezo/ensta).

### Installation
```shell
pip install ensta
```

#### This is how to fetch follower list:
```py
from ensta import Host, NewSessionID

sessionid = NewSessionID("username", "password")

host = Host(sessionid)
followers = host.followers("melon_owe")

for user in followers:
    print(user.username)  # access any data
```

[Check out ensta's github repo for more...](https://github.com/diezo/ensta)

**Comment by AdrianGuretto** [comment]
Have you found any solution to this issue? I have the same problem.

**Comment by diezo** [comment]
Use [Ensta](https://github.com/diezo/ensta).

I already gave the solution a month ago, but it was deleted by the repository owner.

**Comment by hdem607** [comment]
Same here. 243 followers and then error 'Wait a few minutes before try again', despite I am using [30,45] random time between requests.

**Comment by jotauses** [comment]
> Using [Ensta](https://github.com/diezo/ensta), I managed to fetch 2,000 followers, then I stopped the program.
> 
> I already gave the solution a month ago, but it was deleted by the repository owner.
> 
> Install Ensta by running: `pip install ensta --upgrade`
> 
> Here's the code to fetch followers:
> 
> ```python
> from ensta import AutoHost
> 
> host = AutoHost(username, password, file="session.txt")
> followers = host.followers("cristiano")
> 
> for user in followers:
>     if user is None:
>         print("Something went wrong.")
>     else:
>         print(user.username)
> ```

It doesn't work, if you use what you say or the same example from the ENSTA documentation, the number of followers seems to be random, each time it is executed it returns different and repeated followers.

**Comment by subzeroid** [comment]
Fixed and released in instagrapi 2.9.5.

Root cause: authenticated `user_followers(...)` could hit Instagram's private followers endpoint with a large `count=amount`, or receive a deliberately limited private response with `should_limit_list_of_followers=True` and no cursor. The high-level helper now caps private page `count` at `MAX_USER_COUNT` and intentionally falls back to public GraphQL when the private response is limited and does not satisfy the requested amount.

Links:
- instagrapi PR: https://github.com/subzeroid/instagrapi/pull/2618
- instagrapi release: https://github.com/subzeroid/instagrapi/releases/tag/2.9.5
- instagrapi PyPI: https://pypi.org/project/instagrapi/2.9.5/
- instagrapi Codeberg: https://codeberg.org/subzeroid/instagrapi/src/tag/2.9.5
- aiograpi PR: https://github.com/subzeroid/aiograpi/pull/323
- aiograpi release: https://github.com/subzeroid/aiograpi/releases/tag/1.3.5
- aiograpi PyPI: https://pypi.org/project/aiograpi/1.3.5/
- aiograpi Codeberg: https://codeberg.org/subzeroid/aiograpi/src/tag/1.3.5

Live verification on `45600199849`: `user_followers(..., amount=350)` returned 350 after a limited private page (`count=200`, 49 users, `should_limit_list_of_followers=True`) and public GraphQL fallback. Direct `user_followers_v1(..., amount=350)` now avoids the previous `count=350` 400 and returns Instagram's limited private slice.
